### DISCUSSION TITLE
How do i setup Modoboa behind Nginx reverse proxy?

### DISCUSSION THREAD

**Discussion by xLucipher** [modoboa/modoboa#2419] [Help]
Hello, everyone,

I need help. I don't understand how to run Modoboa behind the Nginx reverse proxy. When installing Modoboa, the SSL certificate is automatically stored on mail.example.com.

The reverse proxy runs on a separate VM under Docker (Nginx Proxy Manager). Integrating the other VMs wasn't a problem at all. However, that doesn't work so well with Modoboa.

Nowhere do I find an entry indicating which port the Web UI is running on.
Can someone help me there? If this is not so easy and more effort is required, I would be happy to receive instructions.

I'm using Ubuntu 20.04 LTS

**Comment by manf0001** [comment]
Have you been able to get it working with a reverse proxy?   I have server A running  nginx as a reverse proxy.   Server B running Apache to host my sites,   and Server C running modoboa (nginx)    I can't seem to figure out how to pass https traffic from Server A to Server C to access the modoboa web components.    But it works fine going to my Apache server

**Reply by manf0001** [reply]
Well shortly after I posted this question, I stumbled upon the solution. 

Here is the code I have for my mail conf file on the nginx reverse proxy

'server {
    listen 443 ssl http2;
    server_name mail.example.com mail.example2.com;
    location / {
        proxy_pass https://<IP of Modoboa server>:443;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header Host      $host;
    }
}
`
Below is how I have it configured for both outside and inside access.    As stated above, I have several servers so I use a DNS server inside my network.

I configured my modoboa mail server as mail.example.com     so with that setup, my mail clients need to access mail.example.com to the mail server, and I have to use the same address for the website.

Internal setup is very easy.    I just configured my internal DNS server to point  mail.example.com to the internal IP of my mail server.   That way I can access the website and mail clients, can access mail and it uses the mail.example.com certs I have for it.   (I originally wanted to try to have the web site work with webmail.example.com but I couldn't quite get it and I didn't want to end up breaking my mail server)

For outside access:   for those of us that only have one IP and multiple servers hosting things, and we don't want to poke holes in our firewalls, we have to resort to using a reverse proxy.  On my setup, I'm running nginx, and I have a separate conf file for each site that I want to access from outside.    I also have a wildcard cert to handle the different domains and subdomains I'm using.

so I created a mail.example.com.conf file, and added the above code in there.   After restarting nginx,  I was able to access the web portal from outside, and it was using the wild card cert I had set up.  (I have another conf file specifically for SSL which has my cert info, and all the other config files use that)

as stated I also have a couple of extra domains, and on my proxy server, those other domains are all in the same cert, so I was able to go to mail.example2.com with out issues as well.  

so internally, it's using the letsencrypt cert for mail.example.com, mail.example2.com and everything just points directly to the Modoboa server.   


While external access  I have port 80/443 going to my reverse proxy in turn will pull up the Modoboa Mail site, using  wild card cert, and the other ports for email, 25,  imap, imaps  etc, is being forwarded directly to the mail server, using the mail certs.

I hope this helps down the line.  :-)
